infra/discovery/src/lib/search.js

```diff
@@ -373,9 +372,13 @@ class Listing {
                     type: 'number',
                     script: {
                       lang: 'painless',
+                      // Note: Script is very defensive at checking existence and validity
+                      // of the data to be robust to possibly malformed listings that are on testing
+                      // environments and could also exist in production.
                       source: `
                         if (params._source.containsKey("price") &&
                           params._source.price.containsKey("amount") &&
+                          params._source.price.amount != null &&
```

Comment:
The `Float.parseFloat` calls below [can throw](https://www.geeksforgeeks.org/float-parsefloat-method-in-java-with-examples/) `NumberFormatException` and `NullPointerException`.  Would be cool to wrap it in try-catch to also handle the possible wrong format exceptions.